packages/ember-principled-forms/tests/dummy/app/components/form-field/template.hbs

```diff
@@ -1,35 +1,83 @@
 {{#docs-demo as |demo|}}
-  {{#demo.example name='eager-principled-form-field.hbs'}}
-    {{principled-form-field
-        type=this.user.name.type
-        value=this.user.name.value
+  {{#demo.example name='eager-required-principled-form-field.hbs'}}
+    {{#principled-form-field
         label='Name'
-        onChange=(action 'handleChange' 'name')
+        type=this.user.name.type
         onInput=(action 'handleInput' 'name')
+        as |field|
     }}
+      <label>
```

Comment:
I take it your idea is we'd wrap this in our own responsive form-field component which would be responsible to do this particular part? 🤔 Seems reasonable enough, if so.

Reply:
Yes, the intention here was to make this component that the package ships as flexible as possible. We can impose our own structure on our project-specific form fields. Even there, however, I would really like us to still support both of the following:

```
{{mwc-form-field ...}}

{{#mwc-form-field ... as |field|}}
    <input ... />
{{/mwc-form-field}}
```

Reply:
Side note: I was heavily inspired by [`downshift`](https://blog.kentcdodds.com/introducing-downshift-for-react-b1de3fca0817). I'm probably biased though because I contributed to the library. I do, however, remain convinced that it's super useful for a library component to impose the minimal amount of HTML possible, and I believe the strategy of supplying props to the wrapped component accomplishes that nicely.

Reply:
Update on this review comment – I believe it's the `has-block` helper now. Will poke around to confirm.